numpy/f2py/__init__.py

```diff
@@ -59,10 +62,32 @@ def compile(source,
         f.write(source)
         f.flush()
 
-        args = ' -c -m {} {} {}'.format(modulename, f.name, extra_args)
-        c = '{} -c "import numpy.f2py as f2py2e;f2py2e.main()" {}'
-        c = c.format(sys.executable, args)
-        status, output = exec_command(c)
+        args = ['-c', '-m', modulename, f.name]
+
+        if os.name == 'posix':
+            posix = True
+        else:
+            posix = False
+
+        extra_args = shlex.split(extra_args, posix=posix)
+
+        if extra_args is not None:
+            args.extend(extra_args)
+
+        c = [sys.executable,
+             '-c',
+             'import numpy.f2py as f2py2e;f2py2e.main()'] + args
+        try:
+            output = subprocess.check_output(c)
+        except subprocess.CalledProcessError as exc:
+            status = exc.returncode
+            output = ''
+        except OSError:
+            # preserve historic status code used by exec_command()
+            status = 127
+            output = ''
+        else:
+            status = 0
         if verbose:
             print(output)
     finally:
```

Comment:
Something for another CR: replace this `try`/`finally` with a `with` statement.

Reply:
With improved % patch diff coverage now, we may be safer to do this -- happy to do so if you prefer, but I guess we may want to contain the amount of disruption in a single PR.

Reply:
> I guess we may want to contain the amount of disruption in a single PR.

Yep, I'd be in favor of doing this later